Tasks.py

```diff
@@ -311,116 +311,125 @@ def perform_rebalance_task(rebalanceTicker, rebalanceSellAmount, eligibleTicker,
-                                logger.debug("Canceling buy order")
+                            logger.info("Buying " + str(elgibleBuyAmount) + " of " + eligibleTicker + " at " + str(eligibleCoinTicker.BTCVal))
+                            if partial_filled == True:
+                                buyOrderUUID = em.create_buy_order(eligibleTicker, partial_fill_amount/eligibleCoinTicker.BTCVal, eligibleCoinTicker.BTCVal)['id']
+                            else:
+                                buyOrderUUID = em.create_buy_order(eligibleTicker, elgibleBuyAmount, eligibleCoinTicker.BTCVal)['id']
+                            time.sleep(60*indexInfo.OrderTimeout)
+
+                        # Check order succeded through
+                        if CondexConfig.DEBUG == True:
+                            logger.debug("Fetching order")
+                            coinBuyIncomplete = False
+                        else:
+
+                            order_result = em.fetch_order(buyOrderUUID)
+                            order_filled_amount = order_result['filled']
+
+                            if order_result['status'] == "closed":
+                                logger.info("Bought coin " + eligibleTicker + " for " + str(order_result['price']))
+                                coinBuyIncomplete = False
+
+                            elif (order_filled_amount*eligibleCoinTicker.BTCVal) > CondexConfig.BITTREX_MIN_BTC_TRADE_AMOUNT and order_result['status'] == "open":
+                                em.cancel_order(buyOrderUUID)
+                                logger.debug("Bought partial of coin " + eligibleCoinTicker + " for " + str(order_result['price']))
+                                coinBuyIncomplete = False
+
                             else:
-                                try:
-                                    em.cancel_order(buyOrderUUID)
-                                except:
-                                    coinBuyIncomplete = False
-                                    pass # order failed to cancel got filled previously
-                                logger.debug("Buy Order Timeout Reached")
-                            time.sleep(10) #Magic Number
+                                coinBuyRetryCount = coinBuyRetryCount + 1
+                                if CondexConfig.DEBUG == True:
+                                    logger.debug("Canceling buy order")
+                                else:
+                                    try:
+                                        em.cancel_order(buyOrderUUID)
+                                    except:
+                                        coinBuyIncomplete = False
+                                        pass # order failed to cancel got filled previously
+                                    logger.debug("Buy Order Timeout Reached")
+                                time.sleep(10) #Magic Number
+
 
+        # Delete the locks
+        if CondexConfig.DEBUG != True:

```

Comment:
put this in a finally block and you can delete the stuff in the except block too